libiocage/lib/Fstab.py

```diff
@@ -139,53 +145,82 @@ def parse_lines(
 
             self.add_line(new_line)
 
-    def read_file(self):
+    def read_file(self) -> None:
         if os.path.isfile(self.file_path):
             with open(self.file_path, "r") as f:
-                self.parse_lines(f.read())
-                f.close()
+                self._read_file_handle(f)
                 self.logger.debug(f"fstab loaded from {self.file_path}")
 
-    def save(self):
-        self.logger.verbose(f"Writing fstab to {self.file_path}")
+    def save(self) -> None:
         with open(self.file_path, "w") as f:
-            f.write(self.__str__())
-            f.truncate()
-            f.close()
+            self._save_file_handle(f)
+            self.logger.verbose(f"{self.file_path} written")
 
-        self.logger.verbose(
-            f"{self.jail.dataset.mountpoint}/fstab written"
-        )
+    def _save_file_handle(self, f) -> None:
+        f.write(self.__str__())
+        f.truncate()
+
+    def _read_file_handle(self, f) -> None:
+        self.parse_lines(f.read())
+
+    def update_and_save(
+        self
+    ) -> None:
```

Comment:
this looks unnecessary spread out… are you sure this won't fit on a single line 😅